### PR TITLE
GitHub Actions Workflow to download and pre-process CHIRPS-GEFS

### PR DESCRIPTION
**Pull request by zackarno**
This work-flow is running successfully when manually run. It does the folliwng:

- download the latest CHIRPS-GEFS leadtimes 1-10 rasters into memory. 
- It then crops these to the area of interest. The cropped rasters are held in memory for processing and also downloaded to the specified gdrive directory.
-  zonal mean is calculated per governorate for each leadtime on the in memory rasters. The resulting object is held in memory for next processing steps, but also downloaded to specified gdrive directory as a csv.

### CODE REVIEW

File: .github/workflows/yem-trigger-monitoring.yaml [hunk 1]
```diff
@@ -6,34 +6,48 @@ name: yem-trigger-monitoring
 on:
   workflow_dispatch:
   schedule:
-    - cron: '25 12 * * *'
+    - cron: '10 09 * * *' # chirps-gefs should be updated 9AM UTC, let's give an extra 10 min
```
Comment (turnerm) [inline .github/workflows/yem-trigger-monitoring.yaml:9]: Ah did we confirm that the timestamps [here](https://data.chc.ucsb.edu/products/EWX/data/forecasts/CHIRPS-GEFS_precip_v12/daily_16day/2023/05/23/) are PST and not UTC?

File: .github/workflows/yem-trigger-monitoring.yaml [hunk 2]
```diff
@@ -6,34 +6,48 @@ name: yem-trigger-monitoring
...
+          r-version: '4.x'
+        # https://github.com/r-spatial/sf/blob/main/.github/workflows/tic-db.yml
+      - name: Install Spatial Libs for Linux
         run: |
-          source(file.path("src", "install_req_r_packages.R"))
-        shell: Rscript {0}
+          sudo apt-get install libgdal-dev libproj-dev libgeos-dev libudunits2-dev
+      - name: Install dependencies
+        run: |
+          sudo apt-get install -y libxml2-dev # Add any additional dependencies here
```
Comment (turnerm) [inline .github/workflows/yem-trigger-monitoring.yaml:26]: I would combine the `apt-get installs`, and also start with an `apt-get update`, like [here](https://github.com/OCHA-DAP/Typhoon-Impact-based-forecasting-model/blob/master/.github/workflows/run-typhoon-model.yml#L48)

Comment (turnerm) [reply .github/workflows/yem-trigger-monitoring.yaml:26]: Do you not need [this](https://github.com/OCHA-DAP/Typhoon-Impact-based-forecasting-model/blob/master/.github/workflows/run-typhoon-model.yml#LL47C1-L47C1) PPA for GDAL etc?

File: .github/workflows/yem-trigger-monitoring.yaml [hunk 3]
```diff
@@ -6,34 +6,48 @@ name: yem-trigger-monitoring
...
           CHD_DS_EMAIL_USERNAME: ${{ secrets.CHD_DS_EMAIL_USERNAME }} 
           CHD_DS_EMAIL_PASSWORD: ${{ secrets.CHD_DS_EMAIL_PASSWORD }} 
           CHD_DS_HOST: ${{ secrets.CHD_DS_HOST }} 
-          CHD_DS_PORT: ${{ secrets.CHD_DS_PORT }} 
-      - name: Run yem-trigger-monitoring 
-        run: |
-          source(file.path("src", "update_trigger.R")) 
-        shell: Rscript {0} 
\ No newline at end of file
+          CHD_DS_PORT: ${{ secrets.CHD_DS_PORT }}
```
Comment (turnerm) [inline .github/workflows/yem-trigger-monitoring.yaml:53]: If you use pre-commit it will fix these newline issues for you automatically

**Review by turnerm — COMMENTED** [review]
Looks good, just a couple of comments around the package installs. (I haven't looked at the R bits, saving that for @PaulineNimo).

But just wondering have you tested it and it works? Without the addition GIS PPA?

File: .github/workflows/yem-trigger-monitoring.yaml (hunk 1 above)
Comment (zackarno) [reply .github/workflows/yem-trigger-monitoring.yaml:9]: yes they are PST

File: .github/workflows/yem-trigger-monitoring.yaml (hunk 2 above)
Comment (zackarno) [reply .github/workflows/yem-trigger-monitoring.yaml]: > I would combine the `apt-get installs`, and also start with an `apt-get update`, like [here](https://github.com/OCHA-DAP/Typhoon-Impact-based-forecasting-model/blob/master/.github/workflows/run-typhoon-model.yml#L48)

good idea - did it here:  dc96d67d7a19123560195f7da57e35832d4b0ae4

> Do you not need [this](https://github.com/OCHA-DAP/Typhoon-Impact-based-forecasting-model/blob/master/.github/workflows/run-typhoon-model.yml#LL47C1-L47C1) PPA for GDAL etc?

apparently not it seems to be running without it

**Comment by zackarno** [comment]
> Looks good, just a couple of comments around the package installs. (I haven't looked at the R bits, saving that for ? @PaulineNimo).

@PaulineNimo 

The important R components are: 

-  `update_trigger.R` - main script activated by GH actions which kicks off the process
- `load_chirps_gefs.R` - run inside update_trigger.R to deal with CHIRPS gefs : loads -> crops->downloads -> zonal stats
-  `email.Rmd` -  rendered from update_trigger.R to generate an html email
- `.depends.R` - also activated by GH actions directly (gets req packages)

**Review by turnerm — APPROVED** [review]
Great!

**Review by PaulineNimo — APPROVED** [review]

**Comment by zackarno** [comment]
just added some feedback received on slack in this commit [91e7f3d](https://github.com/OCHA-DAP/pa-aa-yem-flooding/pull/24/commits/91e7f3d8c6e911818255b5782ebc81551f6472e7)
- minor wording adjustments
- resize plots
- added chirps-gefs data source

going to merge now

**Pull request merged**